src/Component.php

```diff
@@ -141,6 +141,18 @@ protected function loadDefinition(string $definition_file_path) {
     return $component_data;
   }
 
+  /**
+   * Returns the array of variants from the component configuration file.
+   *
+   * @return array
+   *   The loaded component variants.
+   */
+  public function getComponentVariants(): array {
```

Comment:
Since you're in a component class already the method name doesn't need to contain the class name once again.
```suggestion
  public function getVariants(): array {
```

or 
```suggestion
  public function getVariantNames(): array {
```